src/cluster_legacy.c

```diff
@@ -1135,14 +1146,19 @@ void clusterReset(int hard) {
 /* -----------------------------------------------------------------------------
  * CLUSTER communication link
  * -------------------------------------------------------------------------- */
-static clusterMsgSendBlock *createClusterMsgSendBlock(int type, uint32_t msglen) {
-    uint32_t blocklen = msglen + sizeof(clusterMsgSendBlock) - sizeof(clusterMsg);
+static void *createClusterMsgSendBlock(int type, uint32_t msglen) {
+    uint32_t blocklen;
+    if (type == CLUSTERMSG_TYPE_PUBLISH_LIGHT || type == CLUSTERMSG_TYPE_PUBLISHSHARD_LIGHT) {
+        blocklen = msglen + sizeof(clusterMsgSendBlockLight) - sizeof(clusterMsgLight);
+    } else {
+        blocklen = msglen + sizeof(clusterMsgSendBlock) - sizeof(clusterMsg);
+    }
     clusterMsgSendBlock *msgblock = zcalloc(blocklen);
     msgblock->refcount = 1;
     msgblock->totlen = blocklen;
     server.stat_cluster_links_memory += blocklen;
     clusterBuildMessageHdr(&msgblock->msg, type, msglen);
-    return msgblock;
+    return (void *)msgblock;
```

Comment:
Passing around `void *` is not good for type safety. Consider unions or other methods.

Reply:
using the anonymous union helped to avoid this